subprojects/core/src/main/java/org/gradle/api/internal/initialization/transform/services/CacheInstrumentationDataBuildService.java

```diff
@@ -0,0 +1,189 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.initialization.transform.services;
+
+import com.google.common.collect.Sets;
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.internal.initialization.transform.utils.InstrumentationAnalysisSerializer;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.services.BuildService;
+import org.gradle.api.services.BuildServiceParameters;
+import org.gradle.internal.classpath.types.ExternalPluginsInstrumentationTypeRegistry;
+import org.gradle.internal.classpath.types.InstrumentationTypeRegistry;
+import org.gradle.internal.file.FileType;
+import org.gradle.internal.hash.Hasher;
+import org.gradle.internal.hash.Hashing;
+import org.gradle.internal.lazy.Lazy;
+import org.gradle.internal.snapshot.FileSystemLocationSnapshot;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.gradle.api.internal.initialization.transform.utils.InstrumentationTransformUtils.SUPER_TYPES_FILE_NAME;
+
+public abstract class CacheInstrumentationDataBuildService implements BuildService<BuildServiceParameters.None> {
+
+    /**
+     * Can be removed once we actually have some upgrades, but without upgrades we currently can't test this
+     */
+    public static final String GENERATE_CLASS_HIERARCHY_WITHOUT_UPGRADES_PROPERTY = "org.gradle.internal.instrumentation.generateClassHierarchyWithoutUpgrades";
+
+    private final Map<Long, ResolutionData> resolutionData = new ConcurrentHashMap<>();
+    private final Lazy<InjectedInstrumentationServices> internalServices = Lazy.locking().of(() -> getObjectFactory().newInstance(InjectedInstrumentationServices.class));
+
+    @Inject
+    protected abstract ObjectFactory getObjectFactory();
+
+    public InstrumentationTypeRegistry getInstrumentingTypeRegistry(long contextId) {
+        InstrumentationTypeRegistry gradleCoreInstrumentationTypeRegistry = internalServices.get().getGradleCoreInstrumentingTypeRegistry();
+        if (!Boolean.getBoolean(GENERATE_CLASS_HIERARCHY_WITHOUT_UPGRADES_PROPERTY) && gradleCoreInstrumentationTypeRegistry.isEmpty()) {
+            // In case core types registry is empty, it means we don't have any upgrades
+            // in Gradle core, so we can return empty registry
+            return InstrumentationTypeRegistry.empty();
+        }
+        return getResolutionData(contextId).getInstrumentingTypeRegistry();
```

Comment:
🤔 The type is `InstrumentationTypeRegistry` but the method says `InstrumentingTypeRegistry`, consolidate?

Reply:
I see both names used in different places, if you decide to consolidate, make sure to search for all occurrences of the variant you decide to drop.

Reply:
Good catch, definitely needs consolidation. I will fix it, thanks!